app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -316,14 +318,31 @@ export default function createRPCMethodTrackingMiddleware({
         eventProperties.params = transformParams(params);
       }
 
-      trackEvent({
-        event,
-        category: MetaMetricsEventCategory.InpageProvider,
-        referrer: {
-          url: origin,
-        },
-        properties: eventProperties,
-      });
+      if (event === MetaMetricsEventName.SignatureRequested) {
```

Comment:
Minor, is it worth putting new logic in functions so we don't contribute to this monolithic block?

Something like `createSignatureFragment` for example?

Reply:
good idea! Updated https://github.com/MetaMask/metamask-extension/pull/26597/commits/8cb7817a4ec5f661f9ff240db459d498603b8445